[ADD] Add mail_attach_existing_attachment module

adrienpeiffer opened the pull request:
This module was written to add the possibility to add attachments located on the object by sending it by email with the mail compose message wizard

Commit:
[ADD] Add mail attach existing attachment module

pedrobaeza commented:
I think a more descriptive name can be _mail_attach_existing_attachment_.

Another functional question. Why limiting to the attachments of the object? It would be also good to attach any of the existing attachments in the system, don't you think?

It would be good also to have a Usage section explaining that the existing object attachments will be proposed to be included in the mail compose windows through checkboxes.

pedrobaeza commented on mail_attach_attachment/__openerp__.py:
```diff
@@ -0,0 +1,40 @@
+    'author': "ACSONE SA/NV,Odoo Community Association (OCA)",
+    'website': "http://acsone.eu",
+    'category': 'Uncategorized',
+    'version': '8.0.0.0.1',
```
8.0.1.0.0

sbidoul commented:
@pedrobaeza I would leave the name as is, knowing that adding the possibility to add other existing attachments can be added in the same module in another PR.

For now the module is useful as is and covers a useful use case: when the user need to send annexes to an invoices or a sale/purchase order, he can attach the annexes to the object and send them easily later by clicking check boxes in the send email dialog. 

@adrienpeiffer I suggest you add a screenshot in the README.rst. There is an example on how to do this in the mis_builder PR.

pedrobaeza commented:
To add "existing" word is only to clear more the purpose of the module and serves also in case of any future enhancement.

About the enhancement itself, it's enough to put it on the "Known issues / Roadmap" section.

sbidoul commented:
@pedrobaeza ok to add "existing" in the module name.

Commit:
[IMP][mail_attach_existing_attachment] Improve README file and add screenshots

Commit:
[IMP][mail_attach_existing_attachment] Change version format

Commit:
[IMP][mail_attach_existing_attachment] Change module name

adrienpeiffer commented:
Hi @pedrobaeza , @sbidoul ,
I modified the readme file, added screenshots, changed the version numbering and renamed the module.
Thanks for your reviews !

rafaelbn commented:
@adrienpeiffer I think this module must install module "document" if not, how you can attach a new document? I have tested in runbot, and If I don't install document modules I cannot reproduce the functionality. Let me know you opinion.

adrienpeiffer commented:
@rafaelbn Yes, technically the module can run without it. However, to add attachments, we need the document module. I'll add it.

Commit:
[ADD][mail_attach_existing_attachment] Add document module in dependencies

rafaelbn commented:
:+1:  tested in runbot.

pedrobaeza commented on mail_attach_existing_attachment/README.rst:
```diff
@@ -0,0 +1,64 @@
+
+To install this module, you need to:
+
+ * Click on install
```
Don't put initial space

pedrobaeza commented on mail_attach_existing_attachment/README.rst:
```diff
@@ -0,0 +1,64 @@
+
+ * Click on install
+
+Configuration and Usage
```
This is only "Usage" IMO

pedrobaeza commented on mail_attach_existing_attachment/README.rst:
```diff
@@ -0,0 +1,64 @@
+Known issues / Roadmap
+======================
+
+The module only allows the addition of attachments linked to the object.
```
Put `*` at the beginning

pedrobaeza commented on mail_attach_existing_attachment/README.rst:
```diff
@@ -0,0 +1,64 @@
+* Then, by sending the object via email, you can select the attachment added earlier
+
+.. figure:: static/description/ex_mail_compose_message.png
+   :alt: Sends the Purchase Order by email
```
You can add the new Try me on runbot button:

```
.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
   :alt: Try me on Runbot
   :target: https://runbot.odoo-community.org/runbot/205/8.0
```

pedrobaeza commented on mail_attach_existing_attachment/__openerp__.py:
```diff
@@ -0,0 +1,41 @@
+        Adding attachment on the object by sending this one""",
+    'author': "ACSONE SA/NV,Odoo Community Association (OCA)",
+    'website': "http://acsone.eu",
+    'category': 'Uncategorized',
```
I would suggest the same of mail module: `Social Network`

pedrobaeza commented:
Some remarks on the README, but :+1: functional and code review

Commit:
[IMP][mail_attach_existing_attachment] Improve README file

adrienpeiffer commented:
@pedrobaeza it's done ! Thanks !

pedrobaeza commented:
:+1:

hbrunn commented:
:+1:
